test/hotspot/jtreg/gc/TestReferenceRefersToDuringConcMark.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2020, 2021 Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package gc;
+
+/* @test
+ * @requires vm.gc != "Epsilon"
+ * @requires vm.gc != "Shenandoah"
```

Comment:
I think this test works for Shenandoah so long as it's not in IU mode.  Is that possible to exclude with another `@requires` constraint?

Reply:
How would I do that? IU mode can only be distinguished by VM flag.

Reply:
I've not tried it, but this might work.
`@requires vm.gc != "Shenandoah" | vm.opt.ShenandoahGCMode != "iu"`

Reply:
Indeed, it does. I changed the test requires accordingly.